### PR TITLE
deletePrefix: more detailed test

### PR DESCRIPTION
**Pull request by haraldschilly**
related to https://github.com/sagemathinc/local-storage-lru/issues/3

### CODE REVIEW

File: src/local-storage-lru.ts [hunk 1]
```diff
@@ -432,13 +432,9 @@ export class LocalStorageLRU {
...
-    for (let i = 0; i < this.ls.length; i++) {
-      const key = this.ls.key(i);
-      if (key == null) continue;
-      if (key.startsWith(prefix) && key !== this.recentKey) {
-        this.delete(key);
-      }
-    }
+    this.keys()
+      .filter((k) => k.startsWith(prefix) && k !== this.recentKey)
+      .forEach((k) => this.delete(k));
```
Comment (boxymoron) [inline src/local-storage-lru.ts:437]: Add try/catch as mentioned here: https://github.com/sagemathinc/local-storage-lru/issues/3#issuecomment-2189994237

Comment (haraldschilly) [reply src/local-storage-lru.ts:437]: I don't want to catch the error, when deleting `recentKey`. If that fails, it's a problem with the implementation itself. The other operations are caught in the delete method and issue a warning.

Comment (boxymoron) [reply src/local-storage-lru.ts:437]: Ok

**Comment by haraldschilly** [comment]
ok, I've merged this and will make a new version

**Pull request merged**